scripts/process/process.storm_sites.R

```diff
@@ -81,49 +27,25 @@ process.select_flood_sites <- function(viz = as.viz('storm-sites-flood')) {
   gage_nws_flood <- gage_nws_join %>% filter(max_gage > flood.stage | max_forecast > flood.stage) %>% 
     mutate(site_no = paste("nwis", site_no, sep = "-"))
   
-  #filter storm sites
-  storm_sites_filtered <- storm_sites[storm_sites$id %in% gage_nws_flood$site_no,]
-  saveRDS(object = storm_sites_filtered, file = viz[['location']])
-}
-
-#fetch NWIS iv data, downsample to hourly
-
-process.getNWISdata <- function(viz = as.viz('gage-data')){
-  required <- c("depends", "location")
-  checkRequired(viz, required)
-  depends <- readDepends(viz)
-  siteInfo <- depends[['storm-sites']]
-  sites_active <- dplyr::filter(siteInfo@data, class == 'nwis-dot')$id
-  sites_active <- gsub(pattern = "nwis-", replacement = "", x = sites_active)
-  
-  dateTimes <- depends[['timesteps']]$times
-  dateTimes_fromJSON <- as.POSIXct(strptime(dateTimes, format = '%b %d %I:%M %p'), 
-                                   tz = "America/New_York")
-  
-  start.date <-  as.Date(dateTimes_fromJSON[1])
-  end.date <- as.Date(dateTimes_fromJSON[length(dateTimes)])
+  nwis_id <- paste0("nwis-", sites@data$site_no)
   
-  sites_fetcher_params <- getContentInfo('sites')
-  parameter_code <- sites_fetcher_params[['pCode']]
+  library(sp)
+  sites_filtered <- sites[nwis_id %in% gage_nws_flood$site_no, ]
   
-  nwisData <- dataRetrieval::renameNWISColumns(
-    dataRetrieval::readNWISdata(service="iv",
-                                parameterCd=parameter_code,
-                                sites = sites_active,
-                                startDate = start.date,
-                                endDate = end.date,
-                                tz = "America/New_York"))
+  is.featured <- rep(TRUE, nrow(sites_filtered@data))
   
-  nwisData <- dplyr::filter(nwisData, dateTime %in% dateTimes_fromJSON)
+  sites_filtered@data <- data.frame(id = paste0('nwis-', sites_filtered$site_no), 
+                            class = ifelse(is.featured, 'nwis-dot','inactive-dot'),
+                            r = ifelse(is.featured, '3.5','1'),
+                            onmousemove = ifelse(is.featured, sprintf("hovertext('USGS %s',evt);",sites$site_no), ""),
+                            onmouseout = ifelse(is.featured, sprintf("setNormal('sparkline-%s');setNormal('nwis-%s');hovertext(' ');", sites$site_no, sites$site_no), ""),
```

Comment:
Just a note for future us. I think this caused the interaction linkages between sparks and dots to break.

Reply:
I think the issue may also have been complicated by more than one element having the same id. The 'id' attribute value should be unique with the DOM. 

In a somewhat related topic, we may want to think about assigning the event handlers in the javascript. It should lighten the load of the svg (less text).

Reply:
Yeah, agreed. I have something in the retro related to figuring out how to structure the SVG so that life is easier on the other end of things. 

Code committed by me later in the sprint process to deal w/ z-indexing got really sloppy and needs a step back and redesign for the future.